src/lib/import_export/binary/binary_parser.cpp

```diff
@@ -133,6 +133,8 @@ std::shared_ptr<BaseSegment> BinaryParser::_import_segment(std::ifstream& file,
       return _import_value_segment<ColumnDataType>(file, row_count, is_nullable);
     case EncodingType::Dictionary:
       return _import_dictionary_segment<ColumnDataType>(file, row_count);
+    case EncodingType::FixedStringDictionary:
```

Comment:
This should allow us to remove the `default:` here and in the writer, correct?

Reply:
`default` catches and fails if the encoding type in the binary file is invalid, so I think it is okay to keep it.

Reply:
How about moving the fail outside of the switch? That way we still handle invalid encoding types but the compiler complains about missing (valid) encoding types.

Reply:
If we only want to be sure that we crash, we would not need to add anything, as reading the value as EncodingType should fail. If there is no need to add a failure notice (i.e. reading the value as int, check if is not larger than there are values in EncodingType and finally cast it), this could be enough.

Reply:
My point is that removing `default` adds compile-time checking that all encodings are covered

Reply:
Yes, of course, I just wanted to know if we want to have a hint if the EncodingType is correct.

Reply:
You might be overcomplicating this. If the switch case gets an invalid value, it will jump to the end without executing any case. By moving the `Fail` outside the switch statement, that is what is being executed. If you search for `Fail("Invalid enum value");`, you will see this already being done in different places.

Reply:
Yes, this wa stupid :D